src/DeploymentManager.ts

```diff
@@ -0,0 +1,269 @@
+import type http from 'http';
+import type https from 'https';
+import path from 'path';
+import axios from 'axios';
+import type { AxiosInstance, AxiosRequestHeaders, AxiosResponse } from 'axios';
+import fs from 'fs-extra';
+import { InvalidParameterError } from './errors';
+import type LogHandler from './LogHandler';
+
+// #region types
+export interface CreateDeploymentManagerOptions {
+	/**
+	 * Request timeout (ms)
+	 * 0 implies no timeout
+	 * @defaultValue 0
+	 */
+	timeout?: number;
+	/** Optional http agent */
+	httpAgent?: http.Agent;
+	/** Optional https agent */
+	httpsAgent?: https.Agent;
+}
+
+interface DeploymentManagerConstructorOptions {
+	/** Optional http agent */
+	httpAgent?: http.Agent;
+	/** Optional https agent */
+	httpsAgent?: https.Agent;
+}
+
+export interface DeployOptions {
+	/**
+	 * Create directory when deploying
+	 * @defaultValue false
+	 */
+	createDir?: boolean;
+}
+
+interface AuthenticateResult {
+	Cookie: string;
+	'X-XSRF-TOKEN': string;
+}
+
+interface SubroutinesGetValue {
+	[key: string]: unknown;
+	/** Subroutine definition name */
+	name: string;
+}
+type SubroutinesGetResult = Record<string, SubroutinesGetValue>;
+
+interface SubroutineParameterDetail {
+	name: string;
+	type: 'string' | 'json';
+	order: number;
+	dname: '';
```

Comment:
The Swagger definition for this is very poorly documented.  This is for the parameter definitions in the subroutine definition.  I think `dname` has something to do if the parameter is a dynamic array rather than `json`.  I don't know what the valid type is actually supposed to be because the Swagger definition is so lacking.  I had to actually watch the network traffic for what MVIS admin was passing to come up with these definitions.